hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java

```diff
@@ -627,7 +628,16 @@ public Configuration getConfiguration() {
 
   private void checkClosed() throws DoNotRetryIOException {
     if (this.closed) {
-      throw new DoNotRetryIOException(toString() + " closed");
+      throw new ConnectionClosedException(toString() + " closed");
+    }
+  }
+
+  /**
+   * Thrown when connection is closed.
+   */
+  private static class ConnectionClosedException extends DoNotRetryIOException {
+    ConnectionClosedException(String message) {
+      super(message);
```

Comment:
Why is a new ConnectionClosedException defined? The above code already imports one.

Reply:
Otherwise, it looks good.

Reply:
Good question. I didn't even see this class import.

Looking at it, the CCE is no good to me and I think probably no good to anyone since it doesn't subclass DoNotRetryIOE. I left a TODO on that class and called this private local class LocalCCE instead. Thanks @huaxiangsun . Let me update the patch. Will merge if all good.